lib/cpp/src/thrift/server/TConnectedClient.h

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef _THRIFT_SERVER_TCONNECTEDCLIENT_H_
+#define _THRIFT_SERVER_TCONNECTEDCLIENT_H_ 1
+
+#include <boost/shared_ptr.hpp>
+#include <thrift/TProcessor.h>
+#include <thrift/protocol/TProtocol.h>
+#include <thrift/server/TServer.h>
+#include <thrift/transport/TTransport.h>
+
+namespace apache {
+namespace thrift {
+namespace server {
+
+/**
+ * This represents a client connected to a TServer.  The
+ * processing loop for a client must provide some required
+ * functionality common to all implementations so it is
+ * encapsulated here.
+ */
+
+class TConnectedClient : public apache::thrift::concurrency::Runnable {
+  public:
+    /**
+     * Constructor.
+     *
+     * @param[in] serverType     the server type as a string, used
+     *                           for logging output.
+     * @param[in] processor      the TProcessor
+     * @param[in] inputProtocol  the input TProtocol
+     * @param[in] outputProtocol the output TProtocol
+     * @param[in] eventHandler   the server event handler
+     * @param[in] client         the TTransport representing the client
+     */
+    TConnectedClient(const std::string& serverType,
+                     boost::shared_ptr<apache::thrift::TProcessor> processor,
+                     boost::shared_ptr<apache::thrift::protocol::TProtocol> inputProtocol,
+                     boost::shared_ptr<apache::thrift::protocol::TProtocol> outputProtocol,
+                     boost::shared_ptr<apache::thrift::server::TServerEventHandler> eventHandler,
+                     boost::shared_ptr<apache::thrift::transport::TTransport> client);
```

Comment:
This is fine.  I have found the use of references to smart pointers have been confusing to folks with less C++ experience, so I tend to avoid them.  I would hope the compiler could see the constructor is doing nothing and optimize this to a single instantiation, however I have no problem changing it as requested.